gradio/components.py

```diff
@@ -444,15 +433,46 @@ def __init__(
         self.cleared_value = ""
         self.test_input = default_value
         self.interpret_by_tokens = True
-        IOComponent.__init__(self, label=label, css=css, **kwargs)
+        IOComponent.__init__(
+            self,
+            label=label,
+            show_label=show_label,
+            interactive=interactive,
+            css=css,
+            visible=visible,
+            **kwargs,
+        )
 
-    def get_template_context(self):
+    def get_config(self):
         return {
             "lines": self.lines,
             "max_lines": self.max_lines,
             "placeholder": self.placeholder,
             "default_value": self.default_value,
-            **IOComponent.get_template_context(self),
+            **IOComponent.get_config(self),
+        }
+
+    @staticmethod
+    def update(
+        value: Optional[Any] = None,
```

Comment:
There is a discrepancy between the name of the parameter in the constructor `default_value` and `value`. Can we set both to be `value`?

Reply:
I think update method should just follow/respect component's parameters.
Some has `default_value`, some has `choices `and some has no default_value. Furthermore IMO name of default_value is another topic, and not necessary for now to change.

Reply:
Yeah parameters in `update()` should be consistent with constructor parameters.

But we should settle `default_value` vs `value` now I think so that we don't have to worry about backwards compatibility. Imo `value` is a better name than `default_value` if you consider the ways this parameter is used:
* constructing a component in `interactive` mode (e.g. input): here either `value` or `default_value` make sense as this is the original value of the component. E.g. the [text in a HTML textbox](https://www.w3schools.com/jsref/prop_text_value.asp) is called "value"
* constructing a component in  `non-interactive` mode (e.g. output): here `default_value` doesn't make sense, as this value cannot be changed -- it is simply the value that is displayed
* updating a component: again, `default_value` is not ideal because you are updating this value

Not to mention `value` is a shorter name!

Reply:
My main problem is `value` does not map to initial value or default value in my mind. That might just for me, if you think it will translate into the initial value in the community, it can be way to go.

And in all 3 cases `default_value` makes sense for me, 
2. non-interactive has just default value
3. we are updating the default_value of the component